### PR TITLE
Assume periodic and postsubmit testgroups should exist and automatically create them

### PR DESCRIPTION
**Pull request by Katharine**
Currently it is always a presubmit error for a periodic or postsubmit testgroup to not be created.

Instead of requiring people to do that, if they just want a basic group we can just assume it and do it for them. If they want fancier features on their testgroup, they can still manually create it in config.yaml.

Presubmits can opt in by using one of the testgrid annotations, periodics and postsubmits can opt out with the annotation `"testgrid-create-test-group": "false"` (though I can imagine no reason to do so since presubmits will then fail).

Perhaps review by commit: first commit has the code change, second commit removes all the trivial testgroups.

/cc @michelle192837 @spiffxp @fejta

### CODE REVIEW

File: testgrid/config.yaml [hunk 1]
```diff
@@ -51,14 +51,6 @@ default_dashboard_tab:
 #
 
 test_groups:
```
Comment (fejta) [inline testgrid/config.yaml]: Can we have something somewhere to help people understand how all this magic works?

Comment (spiffxp) [reply testgrid/config.yaml]: Perhaps a generated description that links back to the file that resulted in this being generated?

**Comment by Katharine** [comment]
People do still need to make the dashboard, and add the testgroup to the dashboard.

File: testgrid/config.yaml (hunk 1 above)
Comment (Katharine) [reply testgrid/config.yaml]: I can, though I kinda doubt anyone will see it given how absurdly large the file is.

Comment (Katharine) [reply testgrid/config.yaml]: @spiffxp This file isn't generated - the ones remaining here were hand-crafted, and have some special property that prevents us from automatically generating them. Specifying a testgroup here prevents us from automatically creating a new one.

Comment (spiffxp) [reply testgrid/config.yaml]: The descriptions show up in testgrid's UI, which might be a more discoverable way of understanding how/why the tabs still exist

Comment (Katharine) [reply testgrid/config.yaml]: We aren't autogenerating tabs (those are still done manually), so we can't autogenerate descriptions for them.

(We _will_ add a tab if you specify a `testgrid-dashboards` annotation, in which case the default description does explain why it exists.)

Comment (Katharine) [reply testgrid/config.yaml]: To clarify: this only adds _test groups_, which in the majority of cases are regurgitating the same two lines for every job solely to pacify the configurator.

In my experience, the cognitive load of actually figuring out what the two lines should be is remarkably high, and manually writing them does not provide any real benefit.

This PR makes no changes to how dashboards and tabs are specified.

Comment (Katharine) [reply testgrid/config.yaml]: (also, I added a comment explaining.)

**Comment by michelle192837** [comment]
Think this looks good, I'd like to also see the presubmit mentioned note that 1) a test group is automatically created, and 2) that you'll need to add a dashboard ("and you can do it with an annotation, or manually one on in testgrid/config.yaml"). (I'm not sure where it is or what it already says though)

**Comment by Katharine** [comment]
Currently [it just says](https://github.com/kubernetes/test-infra/blob/6f47493c21f0c12ae949ba1ec3530380fee245d2/testgrid/cmd/configurator/config_test.go#L326) `Testgroup %v - defined but not used in any dashboards`

**Comment by Katharine** [comment]
I added a more detailed note to the presubmit.

**Comment by michelle192837** [comment]
Ah, gotcha. Seems good to me.

File: testgrid/cmd/configurator/config_test.go [hunk 2]
```diff
@@ -321,11 +321,17 @@ func TestConfig(t *testing.T) {
...
+	missedTestgroups := false
 	for testgroupname, occurrence := range testgroupMap {
 		if occurrence == 1 {
-			t.Errorf("Testgroup %v - defined but not used in any dashboards", testgroupname)
+			t.Errorf("Testgroup %v - defined but not used in any dashboards.", testgroupname)
+			missedTestgroups = true
 		}
 	}
+	if missedTestgroups {
+		t.Logf("Note: Testgroups are automatically defined for postsubmits and periodics.")
```
Comment (fejta) [inline testgrid/cmd/configurator/config_test.go]: Would like to eventually do it for presubmits too.

Comment (Katharine) [reply testgrid/cmd/configurator/config_test.go]: Agree! We are a bit sketchy on our enforcement of that today, so I stuck to the easy ones.

**Pull request merged**